app/controllers/ansible_repository_controller.rb

```diff
@@ -40,11 +44,21 @@ def button
         page.replace("gtl_div", :partial => "layouts/gtl")
       end
     when "ansible_repository_reload"
+      # @FIXME Should be passed in params by the button. Does not work well with multiple windows.
       params[:display] = @display if @display
-      show
+
+      init_show
+
+      template = 'show'
+      if self.class.display_methods.include?(params[:display])
+        display_nested_list(params[:display])
```

Comment:
how is `display_nested_list` relevant here? Are you introducing any nested lists?

Reply:
@himdel This is a piece taken from the `show` method in a not very DRY way.

The `show` method doesn’t [reach]((https://github.com/ManageIQ/manageiq-ui-classic/blob/master/app/controllers/mixins/generic_show_mixin.rb#L38)) the `custom_display_call` at all, so at least this one would have to remain here. Because of that and some other struggles I decided to leave only what’s necessary here from the `show` method.

That is: `init_show` and those two `if`s. The first is for `playbooks` subpage, which is a nested list, the second is for the new `output` subpage. I am however not sure whether it decreases complexity – the whole `show` method is not called here – or increases it – there are two decision points introduced.

However, since the Refresh button disappeared, this might not be actually needed at all. Moreover, the existing code there could be safely removed.